service/qos/service_level_controller.cc

```diff
@@ -89,12 +90,21 @@ void service_level_controller::set_distributed_data_accessor(service_level_distr
 future<> service_level_controller::stop() {
     // unregister from the service level distributed data accessor.
     _sl_data_accessor = nullptr;
+    try {
+        co_await std::exchange(_distributed_data_updater, make_ready_future<>());
+    } catch (...) {
+        // ignore all exceptions from distributed updater initialization
```

Comment:
Maybe log at least? we could have something broken and not know about it.

Reply:
Yeah, I had the same thing in my mind, though I don't how helpful it will be...
For example: https://jenkins.scylladb.com/view/master/job/scylla-master/job/next/3405/artifact/logs-gating.release.2/1618393293064_deletion_test.RangeDeletionTester.delete_by_1ck_range_less_more_test/node4.log
```
ERROR 2021-04-14 09:41:32,078 [shard 0] init - Unexpected error shutting down service level controller: exceptions::unavailable_exception (Cannot achieve consistency level for cl ONE. Requires 1, alive 0)
```
It won't hurt to WARN about it though.

Reply:
In that case I'd rather go back to catching specific exception types - `exceptions::unavailable_exception` was the one that got detected in dtests. Listing all expected exception types is the best way to ensure that we don't ignore something genuinely broken, even though it might involve a couple of failed dtest runs to recognize all the expected failures.

I'll restore catching specific types in v3 and add the newly detected error.

Reply:
why not log a warning about ignoring the exception here?

Reply:
If the intent is that we want to know if an unexpected exception happens due to a bug, then it's best to just use the existing mechanism - report an unexpected exception on shutdown as an error and investigate. I don't remember how dtests react to warnings, but I think it's not a good idea either way:
 - if warnings also fail the test, then there's no point in having a separate mechanism for reporting it, main.cc already does that
 - if warnings don't fail the test, then we actually wouldn't know when a bug happens, because nobody reads logs from successful dtests

Reply:
> If the intent is that we want to know if an unexpected exception happens due to a bug

But this hunk swallows *all* exceptions, with no indication whatsoever that they happen.
I proposed a warning here since it doesn't cause tests to fail but could come handy in postmortem analysis as extra evidence to something else that went bad.

Reply:
Ooh we're talking about the first one. Sorry, sure, I forgot it existed. Warning makes perfect sense here.